packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -2093,6 +2097,38 @@ class IrisGrid extends Component<IrisGridProps, IrisGridState> {
     });
   }
 
+  async setPartitionConfig(
+    model: PartitionedGridModel,
+    rowIndex: GridRangeIndex
+  ): Promise<void> {
+    const keyTable = await this.pending.add(
+      model.partitionKeysTable(),
+      resolved => resolved.close()
+    );
+    assertNotNull(rowIndex);
+    keyTable.setViewport(rowIndex, rowIndex);
+    try {
+      const data = this.getRowDataMap(rowIndex);
+      const values = Object.entries(data)
+        .filter(([key, value]) =>
+          model.partitionColumns.map(column => column.name).includes(key)
+        )
+        .map(column => column[1].value);
```

Comment:
We can build a set here at the start so we don't need to remap every column table for every key here. Right now this is `O(k*(c^2))` (iterates through each key, then iterates through each column remapping it, then iterates through each column again to check if it matches the key). That's way more complex than required; we don't need to be super optimized as this is only done on a double click, but we can definitely do better than that.
Just create a set of the column names before you do this, e.g.
```suggestion
      const partitionColumnSet = new Set(model.partitionColumns.map(column => column.name));
      const values = Object.entries(data)
        .filter(([key, _]) => partitionColumnSet.has(key))
        .map(([key, columnData]) => columnData.value);
```

Side note - we _could_ have the model also add a `partitionColumnSet` method, but it's not really necessary since we're not iterating over it a lot... but if the time comes then we can add it.

Reply:
Yes this does make sense, definitely a more time-efficient approach. Thanks for the suggestion.